node/main.go

```diff
@@ -0,0 +1,213 @@
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(output)
+}
+
+func handleRestart(w http.ResponseWriter, req *http.Request) {
+	killKit(ctx)
+	initKit()
+
+	resp := map[string]interface{}{"ready": node.Ready}
+	output, err := json.Marshal(resp)
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(output)
+}
+
+func handleUrls(w http.ResponseWriter, req *http.Request) {
+	resp := map[string]interface{}{
+		"ready": node.Ready,
+	}
+
+	nodeUrl, err := MultiAddrToHttp(node.FullNode.ListenAddr)
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+	minerUrl, err := MultiAddrToHttp(node.Miner.ListenAddr)
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+
+	if node.Ready {
+		resp["node_url"] = nodeUrl
+		resp["miner_url"] = minerUrl
+	}
+	output, err := json.Marshal(resp)
+	if err != nil {
+		w.Write([]byte(fmt.Sprintf("%v", err)))
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(output)
+}
+
+func handleKill(w http.ResponseWriter, req *http.Request) {
+	killKit(ctx)
+}
+
+func handleSend(w http.ResponseWriter, req *http.Request) {
+	if !node.Ready {
+		http.Error(w, "the node is not ready yet", 400)
+		return
+	}
+
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		http.Error(w, err.Error(), 400)
+		return
+	}
+
+	var data struct {
+		Receiver string `json:"receiver"`
+		Amount   uint64 `json:"amount"`
+	}
+
+	if err := json.Unmarshal(body, &data); err != nil {
+		http.Error(w, err.Error(), 400)
+		return
+	}
+
+	receiver, err := ethtypes.ParseEthAddress(data.Receiver)
+	if err != nil {
+		http.Error(w, err.Error(), 400)
+		return
+	}
+
+	receiverFilAddr, err := receiver.ToFilecoinAddress()
+	if err != nil {
+		http.Error(w, err.Error(), 400)
+		return
+	}
+	senderAddr, err := node.FullNode.WalletDefaultAddress(ctx)
+	if err != nil {
+		http.Error(w, err.Error(), 400)
+		return
+	}
+
+	msg := &types.Message{
+		From:  senderAddr,
+		To:    receiverFilAddr,
+		Value: types.FromFil(data.Amount),
+	}
+
+	sm, err := node.FullNode.MpoolPushMessage(ctx, msg, nil)
+	if err != nil {
+		http.Error(w, err.Error(), 400)
+		return
+	}
+
+	node.FullNode.WaitMsg(ctx, sm.Cid())
+
+	bal, err := node.FullNode.WalletBalance(ctx, receiverFilAddr)
+	if err != nil {
+		http.Error(w, err.Error(), 400)
+		return
+	}
+
+	resp := map[string]interface{}{
+		"balance": bal.String(),
+	}
+
+	respJson, err := json.Marshal(resp)
+	if err != nil {
+		http.Error(w, err.Error(), 400)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(respJson)
+}
+
+func main() {
+	_ctx, cancel := context.WithTimeout(context.Background(), 10*time.Hour)
+	ctx = _ctx
+	defer cancel()
+
+	http.HandleFunc("/ready", handleReady)
+	http.HandleFunc("/restart", handleRestart)
+	http.HandleFunc("/urls", handleUrls)
+	http.HandleFunc("/kill", handleKill)
+	http.HandleFunc("/send", handleSend)
+
+	go initKit()
```

Comment:
There's some raciness here since in practice we start serving the API before the node has fully initialized. I don't think initKit() needs to be in a goroutine, so you could consider:
1. Returning the (fullnode, miner) tuple from initKit
2. Set the global variables in the main method.
3. _Then_ serve the API.